tools/cythonize.py

```diff
@@ -52,33 +52,29 @@
 # Rules
 #
 def process_pyx(fromfile, tofile):
-    try:
-        from Cython.Compiler.Version import version as cython_version
-        from distutils.version import LooseVersion
-        if LooseVersion(cython_version) < LooseVersion('0.19'):
-            raise Exception('Building %s requires Cython >= 0.19' % VENDOR)
-
-    except ImportError:
-        pass
-
     flags = ['--fast-fail']
     if tofile.endswith('.cxx'):
         flags += ['--cplus']
 
     try:
+        # try the cython in the installed python first (somewhat related to scipy/scipy#2397)
+        from Cython.Compiler.Version import version as cython_version
+    except ImportError:
+        # if that fails, use the one on the path, which might be the wrong version
         try:
-            r = subprocess.call(['cython'] + flags + ["-o", tofile, fromfile])
-            if r != 0:
-                raise Exception('Cython failed')
+            # Try the one on the path as a last resort
+            subprocess.check_call(
+                ['cython'] + flags + ["-o", tofile, fromfile])
```

Comment:
This worries me. A naive developer (me, for instance) wanting to start hacking on NumPy could forget to install Cython into their virtualenv, this could potentially get the one from the OS which may be too old. Couldn't we just remove this?

Reply:
... in favor of `subprocess.call([sys.executable, '-m', 'cython']`

Reply:
I agree this is a little worrisome, but before this patch, it would always do this - at least now if you do create the virtualenv, you get correct behavior.

I don't want to change this in a way that will break existing users building from source in a non-obvious way, which removing the call to the system `cython` would do. Emitting a warning would be reasonable, but I'd like to leave that to a future PR.